testing/checkers.go

```diff
@@ -367,3 +367,87 @@ func CheckBootCmd(obtained, bootCmd string, env []string) error {
 
 	return nil
 }
+
+// The ContainsArray checker verifies that the obtained array contains the expected subarray.
+//
+// For example:
+//
+//     c.Assert([]string{"a", "b", "c"}, ContainsArray, []string{"a", "b"}) # TRUE
+//     c.Assert([]string{"a", "b", "c"}, ContainsArray, []string{"b", "c"}) # TRUE
+//     c.Assert([]string{"a", "b", "c"}, ContainsArray, []string{"b"})      # TRUE
+//     c.Assert([]string{"a", "b", "c"}, ContainsArray, []string{"c", "b", "a"}) # FALSE
+//     c.Assert([]string{"a", "b", "c"}, ContainsArray, []string{"a", "c"}) # FALSE
+//     c.Assert([]string{"a", "b", "c"}, ContainsArray, []string{"x"})      # FALSE
+//     c.Assert([]string{"a", "b", "c"}, ContainsArray, []string{})         # FALSE
+//
+var ContainsArray Checker = &containsArrayChecker{
+	&CheckerInfo{Name: "ContainsArray", Params: []string{"obtained", "subarray"}},
+}
+
+type containsArrayChecker struct {
+	*CheckerInfo
+}
+
+func (checker *containsArrayChecker) Check(params []interface{}, names []string) (result bool, error string) {
+	obtained, err := interface2array(params[0])
+	if err != nil {
+		return false, "Obtained value must be slice."
```

Comment:
Nitpick, but some of the err messages start with capital while others do not. Please, unify them.